### DISCUSSION TITLE
Provide a way to "reset" the stylesheet during development

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [tailwindlabs/tailwindcss#8415] [Ideas]
Usually it's no problem at all during development that the Tailwind stylesheet is only "additive" (in the sense that the JIT engine doesn't remove rules that are no longer found in the `content` files). Sometimes, however, you really do need to get rid of some unused rules, and as far as I know the only way to do that currently is to stop the process and restart it. Such a situation arises _often_ if you're using the Tailwind Typography plugin and customizing the `prose` rules in the config—a deleted rule doesn't "take" without restarting.

It would be extremely helpful to have a mechanism for clearing out the stylesheet (or at least the Typography rules) while the process is running. Depending on the setup, stopping and restarting the whole process can be time-consuming and rather annoying.

**Comment by waj00115** [comment]
Someone can correct me if I'm wrong, but I thought the entire typography plugin / prose class was safelisted because of the unpredictability of needed tags, e.g. a blog constructed of either markdown files or database records that wouldn't be picked up by a scan of the content directories.

**Comment by MichaelAllenWarner** [comment]
@waj00115 

That's not quite right. The `.prose` class and variants like `.prose-lg` aren't _safelisted_, but each one you use gets included in the stylesheet in its entirety (because of the unpredictability you mentioned).

My point is this: if I add a rule to the config for, say, `<dt>` elements in the `.prose` scope, then if I delete that rule from the config, it doesn't disappear from the stylesheet until I stop and restart the whole Tailwind process. That makes experimentation cumbersome.

**Comment by adamwathan** [comment]
Hey! You should be able to do this by saving your `tailwind.config.js` file — we throw out the whole cache any time that happens. If that doesn't happen for you can you provide an example project I can check out?

**Reply by MichaelAllenWarner** [reply]
Thanks for the response. When I get a chance I'll try it in a fresh/minimal repo; maybe there's just something else in my usual setup that's interfering with things in some way.

**Reply by MichaelAllenWarner** [reply]
@adamwathan

I wonder if it's got something to do with how we're using Docker. We've got the Tailwind config set up as a read-only volume, so saving the config on the host machine immediately updates the corresponding config-file in the Docker container. That does successfully trigger the PostCSS pipeline in the Docker container, as it should, but changes to the Typography config in particular (including _adding_ to it, it seems) don't result in observable changes in the browser until I restart the Docker container.

Is there a way I can tell if the cache is getting thrown out? Like a debug mode or something that would notify me in the console when it happens?
